### PR TITLE
Package into a Docker image

### PR DESCRIPTION
**Pull request by itskingori**
Package v2.1 with Alpine 3.4 base image and configure tests and automatic builds.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,18 @@
-# docker-oauth2-proxy
-Docker image for oauth2_proxy, a reverse proxy that provides authentication with Google, Github or other providers
+# Docker OAuth2 Proxy
+
+This is a small docker image for `oauth2_proxy` which is a reverse proxy
+that provides authentication with Google, Github or other providers.
```
Comment (zacblazic) [inline README.md]: Casing of Github should be GitHub.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,18 @@
...
+
+Configure OAuth2 Proxy using config file, command line options, or
+environment variables. See [`bitly/oauth2_proxy` documentation][1] for
+more details.
+
+## Usage
+
+Get a `bash` shell from the container
+
+    $ docker run zappi/oauth2_proxy --version
```
Comment (zacblazic) [inline README.md]: This doesn't get you a shell. 😄

File: README.md (hunk 1 above)
Comment (itsmrwave) [inline README.md]: LOL. I like the OCD in this team.

File: README.md (hunk 2 above)
Comment (itsmrwave) [inline README.md]: Copypasta! 🙈

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  environment:
+    IMAGE_NAME: 'zappi/oauth2_proxy'
+  services:
+    - docker
+
+test:
+  pre:
+    - docker build --tag=$IMAGE_NAME .
+  override:
+    - docker run $IMAGE_NAME --version
```
Comment (zacblazic) [inline circle.yml]: How about pushing to Docker Hub on tag?

Comment (zacblazic) [inline circle.yml]: Oh wait, there's hooks for this on Docker Hub right?

File: README.md (hunk 1 above)
Comment (zacblazic) [inline README.md]: Casing here of Github should be GitHub.

File: README.md (hunk 2 above)
Comment (zacblazic) [inline README.md:30]: This doesn't get us a `bash` shell.

File: circle.yml (hunk 3 above)
Comment (zacblazic) [inline circle.yml]: Do releases in GitHub create tags on Docker Hub for our image?

**Review by zacblazic — CHANGES_REQUESTED** [review]
Few things need to be addressed.

**Review by zacblazic — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (itskingori) [reply README.md]: Fixed in 30f9424.

File: README.md (hunk 2 above)
Comment (itskingori) [reply README.md:30]: Fixed in 9d1db07.

File: circle.yml [hunk 4]
```diff
@@ -12,3 +12,11 @@ test:
     - docker build --tag=$IMAGE_NAME .
   override:
     - docker run $IMAGE_NAME --version
+
+deployment:
+  tagged:
+    tag: /.*/
+    commands:
+      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
```
Comment (zacblazic) [inline circle.yml:20]: Will these environment variables be set up as secrets in Circle CI?

File: circle.yml [hunk 5]
```diff
@@ -12,3 +12,11 @@ test:
...
+      - docker tag $IMAGE_NAME:latest $IMAGE_NAME:$CIRCLE_TAG
+      - docker push $IMAGE_NAME:$CIRCLE_TAG
```
Comment (zacblazic) [inline circle.yml:22]: Does this also push the `latest` tag?

Comment (zacblazic) [reply circle.yml:22]: Ah I see you've added it in another commit. Can they not be pushed together?

Comment (itskingori) [reply circle.yml:22]: I could but I'm not sure it works i.e. does it overwrite? will it error out due to a conflict like ECR?

Comment (itskingori) [reply circle.yml:22]: In [that case](https://github.com/Intellection/docker-oauth2-proxy/pull/1#r79158102) I'd like to revert just that specific change.

File: circle.yml (hunk 4 above)
Comment (itskingori) [reply circle.yml:20]: I dunno. 😰 

I'm also wondering who's details should I use.

File: circle.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    - docker build --tag=$IMAGE_NAME .
+  override:
+    - docker run $IMAGE_NAME --version
+
+deployment:
+  tagged:
+    tag: /.*/
+    commands:
+      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+      - docker tag $IMAGE_NAME:latest $IMAGE_NAME:$CIRCLE_TAG
```
Comment (zacblazic) [inline circle.yml]: Not necessary to put `:latest` here.

**Review by zacblazic — CHANGES_REQUESTED** [review]
Minor change.

File: circle.yml (hunk 5 above)
Comment (zacblazic) [reply circle.yml:22]: I think let's just have it push the Git tag and see what happens first?

File: circle.yml (hunk 6 above)
Comment (itskingori) [reply circle.yml]: Are you sure?

Comment (zacblazic) [reply circle.yml]: Yes, this is how we tag in `deploy.sh`:

```
docker tag $IMAGE_NAME $AWS_ACCOUNT_ID.dkr.ecr.$AWS_REGION.amazonaws.com/$IMAGE_NAME:$VERSION
```

File: circle.yml (hunk 5 above)
Comment (itskingori) [reply circle.yml:22]: > I think let's just have it push the Git tag and see what happens first?

Dropped `latest` push for now. We can do it later.

**Review by zacblazic — APPROVED** [review]
Looks good! 👍

File: circle.yml (hunk 6 above)
Comment (itskingori) [reply circle.yml]: Done in 7619ae6. However, it would have still worked. Docker defaults to `latest` when no tag set.

**Pull request merged**